File: Msys_Programs/MSys_Projects/GoProg/DataStructure/SortingAlgorithm/Bubble_Sort.go
package main

import "fmt"

func BubbleSort(a []int) {
	n := len(a)
	for i := 0; i < n; i++ {
		for j := 0; j < n-i-1; j++ {
			if a[j] > a[j+1] {
				temp := a[j]
				a[j] = a[j+1]
				a[j+1] = temp
			}
		}
	}
}

func main() {
	a := []int{2, 5, 7, 3, 9, 6}
	fmt.Println("Before Sort", a)
	BubbleSort(a)
	fmt.Println("Before Sort", a)
}

File: Msys_Programs/MSys_Projects/PythonProg/Python_DSA/Sorting_Algorithm/BubbleSort.py
def bubblesort(A):
    n = len(A)
    for passes in range(n-1,0,-1):
        for i in range(passes):
            if A[i] > A[i + 1]:
                A[i],A[i+1] = A[i+1], A[i]


A = [4,3,6,7,1,2]
print("Befor Sorting",A)
bubblesort(A)
print("After sorting",A)

# A = [4,3,6,7,1,2]
# n = len(A)
# for passes in range(n-1,0,-1):
#     for i in range(passes):
#         if A[i] > A[i + 1]:
#             temp = A[i]
#             A[i] = A[i + 1]
#             A[i + 1] = temp
#         print("before sort",A)
#     print("after sort",A)